File: VisualBasicAssignment/screen/kangtze/edit.vb
Imports System.Data.SqlClient
Imports System.Drawing.Image
Imports System.Drawing.Imaging
Imports System.Drawing.Bitmap
Imports System.IO
Public Class edit
    Dim ds As DataSet = New DataSet()
    Dim da As SqlDataAdapter


    Private Sub edit_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        Dim cmdstr As String = "select * from food where food_id='" + editid + "'"
        'PictureBox1.Image = Image.FromStream(ms)
        Dim arrImage() As Byte

        If openconnect() = True Then
            ds.Clear()
            da = New SqlDataAdapter(cmdstr, con)
            da.Fill(ds, "Food")

            If ds.Tables("Food").Rows.Count = 1 Then
                arrImage = ds.Tables("Food").Rows(0).Item(2)
                Dim ms As New MemoryStream(arrImage)
                Dim bitmap As New Bitmap(Image.FromStream(ms))


                txtName.Text = ds.Tables("Food").Rows(0).Item(1)
                txtDesc.Text = ds.Tables("Food").Rows(0).Item(3)
                txtPrice.Text = ds.Tables("Food").Rows(0).Item(4)
                lblComboBox.Text = ds.Tables("Food").Rows(0).Item(5)
                pcbProduct.Image = bitmap
            End If
            closeconnect()
        End If
    End Sub

    Private Sub btnCancel_Click(sender As Object, e As EventArgs) Handles btnCancel.Click
        Me.Close()
    End Sub

    Private Sub btnCreate_Click(sender As Object, e As EventArgs) Handles btnCreate.Click

        Dim cmdstr As String = "update [food] set [food_name]=@name, [food_price]=@price, [food_category]=@category  where food_id='" + editid + "'"
        Dim cmd As SqlCommand


        If openconnect() = True Then
            cmd = New SqlCommand(cmdstr, con)
            cmd.Parameters.AddWithValue("@name", txtName.Text)
            cmd.Parameters.AddWithValue("@desc", txtDesc.Text)
            cmd.Parameters.AddWithValue("@price", Integer.Parse(txtPrice.Text))
            cmd.Parameters.AddWithValue("@category", cmbCategory.Text)
            cmd.ExecuteNonQuery()
            closeconnect()
        End If
        Me.Close()
    End Sub
End Class

File: VisualBasicAssignment/screen/kangtze/edit.cs
using System;
using System.Data;
using System.Data.SqlClient;
using System.Drawing;
using System.IO;
using Microsoft.VisualBasic.CompilerServices;

namespace VisualBasicAssignment
{
    public partial class edit
    {
        public edit()
        {
            InitializeComponent();
            _btnCancel.Name = "btnCancel";
            _btnCreate.Name = "btnCreate";
        }

        private DataSet ds = new DataSet();
        private SqlDataAdapter da;

        private void edit_Load(object sender, EventArgs e)
        {
            string cmdstr = "select * from food where food_id='" + Module1.editid + "'";
            // PictureBox1.Image = Image.FromStream(ms)
            byte[] arrImage;
            if (Module1.openconnect() == true)
            {
                ds.Clear();
                da = new SqlDataAdapter(cmdstr, Module1.con);
                da.Fill(ds, "Food");
                if (ds.Tables["Food"].Rows.Count == 1)
                {
                    arrImage = (byte[])ds.Tables["Food"].Rows[0][2];
                    var ms = new MemoryStream(arrImage);
                    var bitmap = new Bitmap(Image.FromStream(ms));
                    txtName.Text = Conversions.ToString(ds.Tables["Food"].Rows[0][1]);
                    txtDesc.Text = Conversions.ToString(ds.Tables["Food"].Rows[0][3]);
                    txtPrice.Text = Conversions.ToString(ds.Tables["Food"].Rows[0][4]);
                    lblComboBox.Text = Conversions.ToString(ds.Tables["Food"].Rows[0][5]);
                    pcbProduct.Image = bitmap;
                }

                Module1.closeconnect();
            }
        }

        private void btnCancel_Click(object sender, EventArgs e)
        {
            Close();
        }

        private void btnCreate_Click(object sender, EventArgs e)
        {
            string cmdstr = "update [food] set [food_name]=@name, [food_price]=@price, [food_category]=@category  where food_id='" + Module1.editid + "'";
            SqlCommand cmd;
            if (Module1.openconnect() == true)
            {
                cmd = new SqlCommand(cmdstr, Module1.con);
                cmd.Parameters.AddWithValue("@name", txtName.Text);
                cmd.Parameters.AddWithValue("@desc", txtDesc.Text);
                cmd.Parameters.AddWithValue("@price", int.Parse(txtPrice.Text));
                cmd.Parameters.AddWithValue("@category", cmbCategory.Text);
                cmd.ExecuteNonQuery();
                Module1.closeconnect();
            }

            Close();
        }
    }
}